### PR TITLE
Upgrade cryptography to address CVE-2023-5678, CVE-2023-6129

### PR DESCRIPTION
**Pull request by gblewis1**
- [x] changelog
- [x] format-python.sh
- [x] integration test
- [x] check BDBA to see if CVE-2023-5678 is still flagged

### CODE REVIEW

**Comment by gblewis1** [comment]
the cryptography library does not yet address this CVE -- see https://github.com/pyca/cryptography/issues/9898. holding until then.

**Comment by nex-maximus** [comment]
## Code Review Analysis

- 🎯 **Main theme:** The PR is focused on upgrading the cryptography library to mitigate a known vulnerability.
- 📝 **PR summary:** The PR updates the cryptography library version in the requirements.txt file to address the security vulnerability CVE-2023-5678. It also includes an entry in the Changelog.md to document this change.
- 📌 **Type of PR:** Bug fix
- 🏅 **Score:** 90
- 🧪 **Relevant tests added:** No
- ⏱️ **Estimated effort to review [1-5]:** 1, because the PR is straightforward, involving a single version bump and a changelog update.
- 🔒 **Security concerns:** No
## Code Review Feedback



<strong>💡 General suggestions:</strong> The PR effectively addresses the security concern by updating the vulnerable package. It is important to ensure that integration tests are run to verify that the new version of the cryptography library does not introduce any regressions or incompatibilities with the existing codebase. Additionally, it would be beneficial to complete the checklist items mentioned in the PR description, such as running integration tests and checking BDBA for the presence of CVE-2023-5678 post-update.


<details> <summary><strong>✨ Usage tips:</strong></summary><hr> 

> Tag me in a comment '@nex-maximus' and add one of the following commands:
> **/review [-i]**: Request a review of your Pull Request. For an incremental review, which only considers changes since the last review, include the '-i' option.
> **/describe**: Modify the PR title and description based on the contents of the PR.
>For example: /review --pr_reviewer.extra_instructions="focus on the file: ..." 

</details>

File: inbm/dispatcher-agent/requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 psutil==5.9.5
 types-psutil==5.9.5.17
-cryptography==41.0.6
+cryptography==41.0.7
```
Comment (nex-maximus) [inline inbm/dispatcher-agent/requirements.txt:5]: Ensure that the updated version of the `cryptography` library is compatible with other dependencies and the overall application. It is recommended to perform thorough testing to confirm that the update does not break any functionality. [important]

File: inbm/Changelog.md [hunk 2]
```diff
@@ -28,6 +28,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
  - dependabot: Bump cryptography from 41.0.4 to 41.0.6 in /inbm/dispatcher-agen (addresses CVE-2023-49083, CVE-2023-5678, CVE-2023-5363)
  - update included reference certifi source code from 2020.12.05 to 2023.7.22, which was not a security issue per se but was flagged in BDBA as it contains CVE-2022-23491 and CVE-2023-37920
  - dependabot: Bump pyinstaller from 5.13.0 to 5.13.1 in all agents/programs (addresses CVE-2023-49797)
+ - RTC 537811: Bump cryptography from 41.0.6 to 41.0.7 in /inbm/dispatcher-agent (addresses CVE-2023-5678)
```
Comment (nex-maximus) [inline inbm/Changelog.md:31]: It is good practice to include the reason for the version bump in the changelog entry, such as mentioning the specific CVE that is being addressed. This provides better context for the change. [medium]

Comment (gblewis1) [inline inbm/Changelog.md:31]: let's combine this description with the 41.0.4 -> 41.0.6 upgrade and batch together all the affected CVEs

Comment (gblewis1) [reply inbm/Changelog.md:31]: already in description

**Comment by nex-maximus** [comment]
**[Persistent review](https://github.com/intel/intel-inb-manageability/pull/462#issuecomment-1890047602)** updated to latest commit https://github.com/intel/intel-inb-manageability/commit/24404fc9d33d805573eb63fe2d343aed8d96548d

File: inbm/dispatcher-agent/requirements.txt (hunk 1 above)
Comment (nex-maximus) [inline inbm/dispatcher-agent/requirements.txt:5]: Ensure that the updated cryptography version does not introduce any breaking changes or incompatibilities with other dependencies. [important]

File: inbm/Changelog.md (hunk 2 above)
Comment (nex-maximus) [inline inbm/Changelog.md:31]: It's good practice to include a hyperlink to the CVE for quick reference and more context about the security vulnerability being addressed. [medium]

**Comment by gblewis1** [comment]
@nmgaston please review (build and integration test passed)

**Review by nmgaston — APPROVED** [review]

**Pull request merged**